remoulade/scheduler.py

```diff
@@ -0,0 +1,153 @@
+import datetime
+import json
+import time
+from typing import Dict, List, Union
+
+import redis
+
+from remoulade import Broker
+
+global_scheduler = None
+
+# TODO: handle timezone
+
+
+def get_scheduler() -> "Scheduler":
+    global global_scheduler
+    if global_scheduler is None:
+        raise ValueError('Scheduler not found, are you sure you called set_scheduler(scheduler) ?')
+    return global_scheduler
+
+
+def set_scheduler(scheduler: "Scheduler"):
+    global global_scheduler
+    global_scheduler = scheduler
+
+
+class ScheduledJob:
+
+    def __init__(self, actor_name: str, args: List = None, kwargs: Dict = None, interval: int = 3600 * 24,
+                 daily_time: datetime.time = None, iso_weekday: int = None, enabled: bool = True,
+                 last_queued: datetime.datetime = None, tz: str = None):
+        if (daily_time is not None and daily_time.tzinfo is not None) or (last_queued is not None and last_queued.tzinfo is not None):
+            raise ValueError("Datetimes should be naive")
+        if daily_time is not None and interval != 3600 * 24:
+            raise ValueError("daily_time should be used with a 24h interval")
+        self.actor_name = actor_name
+        self.interval = interval
+        self.daily_time = daily_time
+        self.iso_weekday = iso_weekday
+        self.enabled = enabled
+        self.last_queued = last_queued
+        self.tz = tz
+        self.args = args if args is not None else []
+        self.kwargs = kwargs if kwargs is not None else {}
+
+    def serialize(self) -> Dict:
+        return {
+            "actor_name": self.actor_name,
+            "interval": self.interval,
+            "daily_time": None if self.daily_time is None else self.daily_time.isoformat(),
+            "iso_weekday": self.iso_weekday,
+            "enabled": self.enabled,
+            "last_queued": None if self.last_queued is None else self.last_queued.isoformat(),
+            "tz": self.tz,
+            "args": self.args,
+            "kwargs": self.kwargs
+        }
+
+    @classmethod
+    def deserialize(cls, data: Dict) -> 'ScheduledJob':
+        return ScheduledJob(
+            actor_name=data["actor_name"],
+            interval=data["interval"],
+            daily_time=None if data["daily_time"] is None else datetime.datetime.strptime(data["daily_time"], '%H:%M:%S.%f').time(),
+            iso_weekday=data["iso_weekday"],
+            enabled=data["enabled"],
+            last_queued=None if data["last_queued"] is None else datetime.datetime.strptime(data["last_queued"], '%Y-%m-%dT%H:%M:%S.%f'),
+            tz=data["tz"],
+            args=data["args"],
+            kwargs=data["kwargs"]
+        )
+
+
+class Scheduler:
+
+    def __init__(self, broker: Broker, schedule: Dict[str, ScheduledJob], *,
```

Comment:
Don't forget to add some doc (unit of period for example). And update the changelog.